### PR TITLE
Dropbox driver ready for implementation with the contributor portal

### PR DESCRIPTION
**Pull request by LucasNVarella**
Only needs some documentation and testing/optimization for production.

Currently authenticates with a personal account,

### CODE REVIEW

File: src/MainPageTB.js [hunk 1]
```diff
@@ -16,6 +16,12 @@ const styles = theme => ({
 });
 
 class MainPageTB extends Component {
+
+    // Testing purposes. Attach this to a button's onClick to see in action - it opens a file attached to the collection, in dropbox.
+    onBioDocumentClick = function(documentName) {
+        this.props.dropbox.openFile(documentName);
```
Comment (matticoli) [inline src/MainPageTB.js]: comment or remove for merge

File: src/App.js [hunk 2]
```diff
@@ -35,14 +36,20 @@ class App extends Component {
...
 
     render() {
         return (
             <div className="App">
                 <ButtonAppBar/>
                 <h1>My Contributions</h1>
-                <MainPageTB contributions={this.state.contributions}/>
+                <MainPageTB 
+                    contributions={this.state.contributions}
+                    dropbox={dbx}
```
Comment (matticoli) [inline src/App.js]: Don't pass singleton as prop, import in child component

File: src/dropbox.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        } else  {
+            config.accessToken = accessToken;
+            this.app = new dropbox.Dropbox(config);
+        }
+    },
+
+    getAccessTokenFromCode: function (redirectUri, code) {
+        this.app.getAccessTokenFromCode(redirectUri, code)
+            .then(function(token) {
+                console.log(token);
```
Comment (matticoli) [inline src/dropbox.js]: don't log access token, this is a security hole. Comment or remove for merge

File: src/dropbox.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    },
+
+    openFile: function (fileName) {
+        window.location.href = "https://www.dropbox.com/home?preview=" + fileName;
+    }
+
+}
+
+function parseQueryString(str) {
+    var ret = Object.create(null);
```
Comment (matticoli) [inline src/dropbox.js]: https://stackoverflow.com/questions/42862253/how-to-parse-query-string-in-react-router-v4

File: src/App.js (hunk 2 above)
Comment (LucasNVarella) [reply src/App.js]: Done

File: src/MainPageTB.js (hunk 1 above)
Comment (LucasNVarella) [reply src/MainPageTB.js]: Done

File: src/dropbox.js (hunk 3 above)
Comment (LucasNVarella) [reply src/dropbox.js]: XD Done

File: src/dropbox.js (hunk 4 above)
Comment (LucasNVarella) [reply src/dropbox.js]: ty, done

**Pull request merged**